src/Apps/Order/Components/BankAccountPicker.tsx

```diff
@@ -62,6 +59,64 @@ export const BankAccountPicker: FC<Props> = props => {
 
   const { submitMutation: setPaymentMutation } = useSetPayment()
 
+  /**
+   * This hook is responsible for setting the bankAccountSelection state
+   * properties that are used by this component to display an appropriate
+   * payment details section to users.
+   */
+  useEffect(() => {
```

Comment:
That's a good refactor, in Order app a lot of routes `index` files have tons of useEffect and other callbacks making it super chunky, and hard to read and maintain.
See how clearer it is right now, thanks for going down this rabbit hole!